### PR TITLE
Target: Add ability to repeat sending of packets

### PR DESCRIPTION
**Pull request by mistressofjellyfish**
In scenarios where one has to hit a certain time window to send a specific packet, it is often times helpful to just flood the target with theses packets.

This PR introduces the concept of a repeater, either count- or time based to a target. It repeats either for a given count or for a given time. The default is unchanged - no repeater at all.

Still to do:

- [x] Unit tests -- I could not find an appropriate spot to add them, as the Target Class seems to be largely untested
- [x] Docs

### CODE REVIEW

File: boofuzz/repeater.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+import time
+
+
+class Repeater(ABC):
```
[the remark this answers is not in the document]
Comment (mistressofjellyfish) [reply boofuzz/repeater.py]: Ah, overlooked that. However, I feel we should start discussing dropping python 2 support as the official support end from cpython is around the corner; let's discuss that in #370.

File: boofuzz/sessions.py [hunk 2]
```diff
@@ -153,7 +156,13 @@ def send(self, data):
         if self._fuzz_data_logger is not None:
             self._fuzz_data_logger.log_info("Sending {0} bytes...".format(len(data)))
 
-        num_sent = self._target_connection.send(data=data)
+        if self.repeater is not None:
+            self.repeater.start()
+            while self.repeater.repeat():
+                num_sent = self._target_connection.send(data=data)
```
[the remark this answers is not in the document]
Comment (mistressofjellyfish) [reply boofuzz/sessions.py]: Good idea. I've added that directly to the repeater class though as it felt more appropriate there.

**Comment by mistressofjellyfish** [comment]
Thanks for the review! I updated my PR to reflect the requested changes. As for the use-case, it is most useful on connections that hop channels (e.g. bluetooth) or rely on the fuzzed party actively scanning (802.11) while using a raw socket to send stuff.

**Comment by SR4ven** [comment]
@mistressofjellyfish I opened https://github.com/mistressofjellyfish/boofuzz/pull/1 including my suggestions.
I felt like this is easier than reviewing and you implementing the changes.

**Comment by mistressofjellyfish** [comment]
Looks good! I'm still a bit concerned that if the testing thread gets descheduled during time.sleep() the test maybe fails, but even if I generated some load and tested it worked fine, so I guess if that ever happens we fix it / re-run the test and have happy fast tests in the meantime.

**Comment by jtpereyda** [comment]
Wow this is pretty cool! Thank you @mistressofjellyfish and @SR4ven . @mistressofjellyfish are you using this with single-message or multiple-message fuzz definitions?

**Comment by mistressofjellyfish** [comment]
Both, actually. I found it most useful though when fuzzing something like 802.11 probe requests. The receivers (while scanning for networks) hop channels, and to ensure your packet gets seen by the target you have to send it while it's listening on your particular channel.

Then you can either play by the rules of the spec or just flood the spectrum. [Charli XCX has summed up my position on this nicely.](https://www.youtube.com/watch?v=ABhDiXbUaBE)

**Pull request merged**